elf/compat.go

```diff
@@ -0,0 +1,65 @@
+// +build linux
+
+// (c) 2018 Suchakra Sharma <suchakra@shiftleft.io>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elf
+
+import (
+	"io/ioutil"
+	"regexp"
+	"runtime"
+)
+
+const defaultSymFile = "/proc/kallsyms"
+
+func GetSyscallFnName(name string) (string, error) {
```

Comment:
Since it is an exported function, could you document it?

Say that the `name` parameter is the bare name of the syscall such as `open` and it returns the name of the kernel function implementing it such as `sys_open`. Maybe rename the parameter to `syscallName` to directly document it in code.